### PR TITLE
Allow keep binary data in Maker Note

### PR DESCRIPTION
**Pull request by clsung**

### CODE REVIEW

**Comment by delphi329** [comment]
for local photos only or for all photos? For htc phones only or for all phone?

**Comment by clsung** [comment]
For all photos which has maker note data inside, according to current libexif implementation, 

1. current EXIF viewer will not show content of MakerNote, for example: libexif shows:
    xxxx bytes undefined data 
    Ref: http://libexif.sourceforge.net/internals/exif-entry_8c-source.html line 462 and line 1739
2. for vendors, they may provide customized parser:
    Ref: http://libexif.sourceforge.net/internals/mnote-canon-entry_8c-source.html
    Ref: http://libexif.sourceforge.net/internals/mnote-fuji-entry_8c.html
3. therefore, I decide to keep those data unmodified, which means we'll do base64 encode on the whole maker note data. So for htc maker, they should decode and parse the data on their side.

File: main_c.c [hunk 1]
```diff
@@ -48,8 +48,18 @@ void import_entry(ExifEntry* entry, void* user_data) {
   ExifIfd ifd = exif_entry_get_ifd(entry);
 
   strncpy(value->name, exif_tag_get_name_in_ifd(entry->tag, ifd), EXIF_VALUE_MAXLEN);
-
-  strncpy(value->value, exif_entry_get_value(entry, exif_text, EXIF_VALUE_MAXLEN), EXIF_VALUE_MAXLEN);
+  if (0x927c == entry->tag) {
+    // length of maker note may exceed EXIF_VALUE_MAXLEN
+    if (entry->size > EXIF_VALUE_MAXLEN) {
+	  value->value = (char *)realloc(value->value, sizeof(char)*entry->size);
```
Comment (imtaco) [inline main_c.c]: should you check if entry->size is in reasonable size ?!

Comment (clsung) [inline main_c.c]: add a max len: 4096

**Pull request merged**